src/modules/checkbox/checkbox.ts

```diff
@@ -0,0 +1,64 @@
+class SmCheckboxDirective implements ng.IDirective {
+
+  static instance(): ng.IDirective {
+    return new SmCheckboxDirective();
+  }
+
+  restrict = 'E';
+  require = '?ngModel';
+  transclude = true;
+  replace = true;
+  template=  '<div class="ui checkbox">' +
+              '<input type="checkbox">' +
+              '<label></label>' +
+            '</div>';
+
+  link = (scope, element, attrs, ngModel, transclude) => {
+    let checked = false;
+    let disabled = false;
+    let input = element.find('input');
+
+    transclude(scope, (nodes) => {
+      element.find('label').append(nodes);
+    });
+
+    element.on('click', toggleFn);
+
+    if (!ngModel) {
+      throw new Error('Semantic-UI-Angular: The \'smCheckbox\' directive requires a \'ng-model\' value');
```

Comment:
We would need to create a generic solution for throwing errors, this is fine for now